instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingConsumer.java

```diff
@@ -98,7 +101,7 @@
 
         // If we extracted neither a trace context, nor request-scoped data (extra),
         // and sharing trace is enabled make or reuse a span for this topic
-        if (extracted.equals(TraceContextOrSamplingFlags.EMPTY) && singleRootSpanOnReceiveBatch) {
+        if (extracted.equals(emptyExtraction) && singleRootSpanOnReceiveBatch) {
```

Comment:
this is the code that tests now caught as a bug. Basically, when we had an empty incoming context, we'd still split into traces for each message. The reason is that we don't have an abstract notion to say if "extra" are empty or not. Meanwhile, the trick to just make an empty context works fine.